ducktape/command_line/main.py

```diff
@@ -56,6 +57,7 @@ def parse_args():
                              "This is primarily useful for test developers who want to interact with running " +
                              "services after a test has run. ")
     parser.add_argument("--version", action="store_true", help="display version")
+    parser.add_argument("--inject", action="store", help="Inject these arguments into the specified test. Use JSON to format.")
```

Comment:
I can think of two possible ways to handle the general problem of running parameterized tests. One is what you're doing here, by specifying the arguments directly. I assume the reason for using JSON is that it's the only way to preserve the type information necessary to pass the arguments correctly.

The second approach is to have the test loader generate all the parameterized test names and then just look them up by name. For example `kafkatest.tests.copycat_test.CopycatStandaloneFileTest.test_file_source_and_sink.converter=org.apache.kafka.copycat.json.JsonConverter.schemas=True` is the full test name in one of the Kafka tests. Since we already have the test loader using filenames, the actual name passed on the command line would be something more like `kafkatest/tests/copycat_test.py::CopycatStandaloneFileTest.test_file_source_and_sink.converter=org.apache.kafka.copycat.json.JsonConverter.schemas=True`.

Any particular reason for going with the former approach? The latter is what I initially imagined, mainly because I figured it would be easy to copy/paste the test name from log files. I guess one reason the arguments approach is nice is that you can run variants of the test that aren't already listed?